chalk-integration/src/program_writer.rs

```diff
@@ -0,0 +1,980 @@
+use chalk_solve::display::{RenderAsRust, WriterState};
+
+use crate::program::Program;
+
+pub trait WriteProgram {
```

Comment:
I'm not really clear on what the role of this trait is

Reply:
It's purely for the tests, and should be removed as we create the logging database